src/AuthContext.js

```diff
@@ -0,0 +1,35 @@
+import React, { createContext, useState, useEffect, useContext } from 'react';
+
+export const AuthContext = createContext(null);
+export const useAuth = () => useContext(AuthContext);
+
+export const AuthProvider = ({ children }) => {
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+    useEffect(() => {
+        const checkLoggedInStatus = () => {
+            const storedData = localStorage.getItem('memestr');
+            setIsLoggedIn(!!storedData);
```

Comment:
What is `!!`?

Reply:
basically we use this "!!"  as  is a shorthand way to convert a value to its corresponding boolean equivalent